src/users/user.service.ts

```diff
@@ -40,6 +41,59 @@ export class UserService {
     return { data: users, total };
   }
 
+  async getUserBySearch(searchUserDto: SearchUserDto): Promise<{ data: User[]; total: number }> {
+    let users = [];
+    let optionalQuery = {};
+    let splitName: string[];
+    let qFirstName, qLastName, qName, qEmail: RegExp;
+    const { user_id, email, name, limit = 0, offset = 0, sort, desc } = searchUserDto;
+    const sorting = sort ? { [sort]: desc } : { createdAt: -1 };
+    // eslint-disable-next-line prefer-const
+    qEmail = new RegExp(`.*${email}.*`, "i");
+    if (name && name.includes(" ")) {
+      splitName = name.split(" ");
+      qFirstName = new RegExp(`.*${splitName[0]}.*`, "i");
+      qLastName = new RegExp(`.*${splitName[1]}.*`, "i");
```

Comment:
if name include multiple spaces, how to solve it?

Reply:
In registration process of accounts, space is forbidden in firstName and lastName (one card was created before to do this); thus it should not include extra space in the name.